src/routes/search/index.js

```diff
@@ -1,517 +1,854 @@
-import { h, Component } from 'preact'
-import { route } from 'preact-router'
-import style from './style'
+import { h, Component } from "preact";
+import { route } from "preact-router";
+import style from "./style";
 // import DisqusThread from '../../components/disqus/DisqusThread';
-import axios from 'axios'
-import classnames from 'classnames'
-import Rodal from 'rodal'
+import axios from "axios";
+import classnames from "classnames";
+import Rodal from "rodal";
 
-import 'rodal/lib/rodal.css'
+import "rodal/lib/rodal.css";
 
 export default class Search extends Component {
-  show (val) {
-    return e => {
-      e.preventDefault()
-      this.setState({ visible: true })
-      this.setState({ currentLocation: val })
-    }
-  }
-  showsuggest () {
-    return e => {
-      e.preventDefault()
-      this.setState({ visiblesuggest: true })
-    }
-  }
-  showdeletion (val) {
-    return e => {
-      e.preventDefault()
-      this.setState({ visibledelete: true })
-      this.setState({ currentLocation: val })
-    }
-  }
-  showDisqus (val) {
-    return e => {
-      e.preventDefault()
-      this.setState({ currentLocation: val })
-      this.setState({ showdisqus: true })
-    }
-  }
-  closeDisqus () {
-    return e => {
-      e.preventDefault()
-      this.setState({ currentLocation: null })
-      this.setState({ showdisqus: false })
-    }
-  }
-  hide () {
-    this.setState({ visible: false })
-    this.setState({ currentLocation: null })
-  }
-  hidesuggest () {
-    this.setState({ visiblesuggest: false })
-  }
-  hidedeletion () {
-    this.setState({ visibledelete: false })
-    this.setState({ currentLocation: null })
-  }
-  handleChange (event) {
-    this.setState({ value: event.target.value })
-  }
-  handleSubmit (event) {
-    this.setState({ loading: false })
-    mixpanel.track('User searched for: ' + this.state.value)
-    axios.get('https://api.govote.org.ng/search?query=' + this.state.value + '&key=k9ihbvse57fvsujbsvsi5362WE$NFD2')
-      .then(res => {
-        this.setState({ loading: true })
-        this.setState({ locations: res.data.data })
-        this.setState({ result: true })
-        route(`${this.props.path}?query=${this.state.value}`)
-      })
-      .catch(err => {
-        console.error(err)
-      })
-    event.preventDefault()
-  }
+	show(val) {
+		return e => {
```

Comment:
Why are you returning a function here? and for `showsuggest`, `showdeletion`, `showDisqus` and `closeDisqus`

Reply:
haven't touched that side of the code, still the original stuff.
@yomete can answer this better